### PR TITLE
*:update dep and mirror fix

### PR DESCRIPTION
**Pull request by csuzhangxc**
many repos depend on TiDB are using a newer version, so update this repo to ensure can be used together with other repos.

NOTE: also mirror fix some code to match the latest pingcap/parser.

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@you06 @zhouqiang-cl PTAL

File: dml_ast.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func (s *SQLSmith) selectStmt(depth int) ast.Node {
 
 func (s *SQLSmith) selectForUpdateStmt(depth int) ast.Node {
 	node := s.selectStmt(depth)
-	node.(*ast.SelectStmt).LockTp = ast.SelectLockForUpdate
+	node.(*ast.SelectStmt).LockInfo.LockType = ast.SelectLockForUpdate
```
Comment (csuzhangxc) [inline dml_ast.go:52]: pingcap/parser updated in https://github.com/pingcap/parser/pull/997.

**Review by mahjonp — APPROVED** [review]
lgtm

**Review by you06 — APPROVED** [review]
LGTM

**Pull request merged**